src/AsmResolver.PE/DotNet/Metadata/SerializedMetadata.cs

```diff
@@ -89,8 +104,15 @@ protected override IList<IMetadataStream> GetStreams()
             if (_streamHeaders.Length == 0)
                 return base.GetStreams();
 
+            var flags = MetadataStreamReaderFlags.None;
+            if (IsEncMetadata)
+                flags |= MetadataStreamReaderFlags.IsEnc;
```

Comment:
This has a data race, as `IsEncMetadata` could be set before the streams are initialized. We probably want to add a `_isEncMetadata` field that stays private to the `SerializedMetadata` class,  similar to `_hasJtdStream`.